contracts/BalanceTrackerBase.sol

```diff
@@ -39,8 +39,7 @@ error TransferFailed(address token, address from, address to, uint256 amount);
 
 abstract contract BalanceTrackerBase {
     event RequesterBalanceAdjusted(address indexed requester, uint256 deliveryRate, uint256 balance);
-    event MechBalanceAdjusted(address indexed mech, uint256 deliveryRate, uint256 balance);
-    event MechPaymentCalculated(address indexed mech, uint256 indexed requestId, uint256 deliveryRate, uint256 rateDiff);
```

Comment:
One event is enough, otherwise they duplicate each other

Reply:
Yes, there are Deposit() events for subscription-like deposit and for deposits on the go. There are events for adjusting mech balances and separately for adjusting requester balances.